File: examples/sample.py
from dataclasses import dataclass

@dataclass
class District:
    name: str
    power: int = 0xFCEE0A

    def glow(self, level: int = 5) -> str:
        # jack into the grid
        return f"{self.name}: {'online' if level > 0 else 'offline'}"

tygers = District("japantown")
print(tygers.glow())

File: examples/sample.ts
// Night City grid bootstrap
import { Grid } from "night-city";

class District {
  readonly name: string;
  power = 0xfcee0a;
  constructor(name: string) {
    this.name = name;
  }
  glow(level = 5): string {
    return `${this.name}: ${level > 0 ? "online" : "offline"}`;
  }
}

const tygers = new District("japantown");
console.log(tygers.glow());
